vulnerabilities/dos-revert.md

```diff
@@ -87,6 +87,15 @@ It's important to take caution in enforcing expected contract balances of tokens
 
 Consider, for example, a contract which expects the Ether balance to be 0 for the first deposit to allow for custom accounting logic. An attacker may forcibly send Ether to the contract before the first deposit, causing all deposits to revert. 
 
+### Divide by zero
+In solidity if the contract attempts to perform division when the denominator is ``zero``, the whole transaction reverts. Thus, the denominator should be always checked before division to prevent DoS revert.
+```solidity
+function foo(uint num, uint den) public pure returns(uint result) {
+  if(den == 0) return 0; // if denominator is 0, return 0 instead of reverting
+  result = num / den;
+}
+```
```

Comment:
IMO it would be better to show the failing case here, i.e. don't check the denominator but then include a comment saying that if the denominator is 0 then execution reverts. My reasoning for this is that we don't necessarily want to return 0 and this could give a false sense of security in taking that option.

Reply:
Fixed in the latest commit!